newlib/libc/tinystdio/conv_flt.c

```diff
@@ -394,8 +394,12 @@ conv_flt (FLT_STREAM *stream, FLT_CONTEXT *context, width_t width, void *addr, u
 	    if (!isdigit (edig))
             {
                 scanf_ungetc(edig, stream, context);
-                if (esign != EOF)
-                    scanf_ungetc(esign, stream, context);
+                if (esign != EOF) {
```

Comment:
The goal of this code is to un-wind the input back to before the exponent. It doesn't completely succeed as the 'p' or 'e' is lost. But, it does manage to restore the invalid exponent digit and sign character (if any). I can't tell what this change is trying to achieve, but putting a 'getc' in the middle doesn't make any sense to me.